modules/programs/mu.nix

```diff
@@ -12,12 +12,14 @@ let
   # Takes the list of accounts with mu.enable = true, and generates a
   # command-line flag for initializing the mu database.
   myAddresses = let
-    # List of account sets where mu.enable = true.
+    # Set of email account sets where mu.enable = true.
     muAccounts =
       filter (a: a.mu.enable) (attrValues config.accounts.email.accounts);
     addrs = map (a: a.address) muAccounts;
-    # Prefix --my-address= to each account's address with mu.enable.
-    addMyAddress = map (addr: "--my-address=" + addr) addrs;
+    # Construct list of lists containing email aliases, and flatten
+    aliases = flatten (map (a: a.aliases) muAccounts);
+    # Prefix --my-address= to each account's address AND all defined aliases
+    addMyAddress = map (addr: "--my-address=" + addr) (addrs ++ aliases);
```

Comment:
This is getting complicated enough that a test feels useful. Not sure quite how to instrument one effectively, though. Maybe @rycee or @berbiche have ideas about how to do that (or maybe they disagree and we don't need a test).

Reply:
I agree that it is getting slightly more complicated. However, mu does not use a configuration file, only a set of command-line arguments passed during the initial index. I am not sure if home-manager could support that particular use. The command _is_ written out to the activation script, so that could be checked...

Reply:
A test would be a nice to have IMO.

Reply:
Agreed it would be nice. I am just not sure how to test it, because `mu` does not write a config file.

Reply:
I assume that `mu init` sets edits some files somewhere. Can we check the contents of those files?

Reply:
The only file that `mu init` creates is an empty database file with records containing the information you provide through the command line. Once you run `mu init`, you run `mu index` to populate that database with the relevant information.

The only file that is edited by this module is the activation script home-manager uses, by inserting an action to create the database.